thunder/tests/test_transforms.py

```diff
@@ -497,3 +497,98 @@ def forward(self, x):
+                return
+
+        err = f"Expected to find symbol {sym} with arguments {arg_vals} in execution trace but didn't find any."
+        raise RuntimeError(err)
+
+    from thunder.transforms.constant_folding import ConstantFolding
+
+    def forward():
+        const_t = torch.tensor([2])
+        getitem = (const_t * 2)[0]
+        return (getitem, const_t)  # (4, [2])
+
+    jforward = thunder.jit(forward, transforms=[ConstantFolding()])
+    actual = jforward()
+    expected = forward()
+    torch.testing.assert_close(actual, expected)
+    exec_trace = thunder.last_traces(jforward)[-1]
+    assert_in_trace(exec_trace, "tensor", ([2],))
+    assert_in_trace(exec_trace, "full", ((), 4))
+
+    def forward(x):
+        const_t = torch.tensor([2])
+        getitem = const_t[0]  # 2
+        getitem_2 = (
+            torch.zeros(
+                2,
+            )
+            + 1
+        )[
+            0
+        ]  # 1
+        return x + getitem + getitem_2
+
+    jforward = thunder.jit(forward, transforms=[ConstantFolding()])
+    x = torch.randn(3, 3)
+    actual = jforward(x)
+    expected = forward(x)
+    torch.testing.assert_close(actual, expected)
+    exec_trace = thunder.last_traces(jforward)[-1]
+    assert_in_trace(exec_trace, "full", ((), 2))
+    assert_in_trace(exec_trace, "full", ((), 1.0))
+
+    def forward(x):
+        const_t = torch.tensor([2], dtype=torch.float16)
+        ones_t = torch.ones(1, dtype=torch.float32)
+        s1 = const_t * 2  # 4
+        s2 = const_t / 1  # 2
+        s3 = s1 * s2 + 10  # 18
+        ones_mul_10 = ones_t * 10  # 10
+        return x[0, 0] + s3 + ones_mul_10
+
+    jforward = thunder.jit(forward, transforms=[ConstantFolding()])
+    x = torch.randn(3, 3)
+    actual = jforward(x)
+    expected = forward(x)
+    torch.testing.assert_close(actual, expected)
+    exec_trace = thunder.last_traces(jforward)[-1]
+    assert_in_trace(exec_trace, "tensor", ([18.0],))
+    assert_in_trace(exec_trace, "tensor", ([10.0],))
+
+    # Constant folding of Python constants.
+    def forward(x):
+        t = torch.tensor(2.0)
+        return x + (t.item() + (t + 1).item())
+
+    jforward = thunder.jit(forward, transforms=[ConstantFolding()])
+    x = torch.randn(3, 3)
+    actual = jforward(x)
+    expected = forward(x)
+    torch.testing.assert_close(actual, expected)
+    exec_trace = thunder.last_traces(jforward)[-1]
+    # exec_trace will look something like this
+    # def computation(x):
+    #     # x: "cpu f32[3]"
+    #     t5 = torch.add(x, 5.0, alpha=1)  # t5: "cpu f32[3]"
+    #         # t5 = ltorch.add(x, 5.0, alpha=1)  # t5: "cpu f32[3]"
+    #         # t5 = prims.add(x, 5.0)  # t5: "cpu f32[3]"
+    #     return t5
+
+    # So we check that torch.add has 5.0 in it's arguments.
+    for bsym in exec_trace.bound_symbols:
+        if bsym.sym.id == "add":
+            assert bsym.args[1] == 5.0
+            break
+    else:
```

Comment:
I don't know. I guess it's OK for the test, but it's one of the things that you have to think a lot what it actually does.